constellations/gemini-admin/src/programs/programEdit.js

```diff
@@ -161,30 +147,98 @@ export class ProgramEditPage extends React.Component {
+                    validators={[
+                        {
+                            validate: (programId) => programId != "",
+                            message: "You must input a programId",
+                        },
+                    ]}
                 />
-                <h4>Program Name</h4>
-                <input
+
+                <TextInput
+                    label="Program Name"
+                    isTextBox={false}
                     value={this.state.inputProgramName}
-                    onChange={(e) => this.handleChange(e, "inputProgramName")}
+                    onChangeCallback={(e) =>
+                        this.handleChange(e, "inputProgramName")
+                    }
+                    required={true}
+                    description="Enter the program name"
+                    validators={[
+                        {
+                            validate: (name) => name != "",
+                            message: "You must input a name",
+                        },
+                    ]}
                 />
-                <h4>Grade1</h4>
-                <input
+
+                <TextInput
+                    label="Grade1"
+                    isTextBox={false}
                     value={this.state.inputGrade1}
-                    onChange={(e) => this.handleChange(e, "inputGrade1")}
+                    onChangeCallback={(e) =>
+                        this.handleChange(e, "inputGrade1")
+                    }
+                    required={true}
+                    description="Enter the lower grade"
+                    validators={[
+                        {
+                            validate: (grade1) => grade1 != "",
+                            message: "You must input a grade",
+                        },
+                        {
+                            validate: (grade1) => parseInt(grade1) >= 1,
+                            message: "Grade cannot be less than 1",
+                        },
+                    ]}
                 />
-                <h4>Grade2</h4>
-                <input
+
+                <TextInput
+                    label="Grade2"
+                    isTextBox={false}
                     value={this.state.inputGrade2}
-                    onChange={(e) => this.handleChange(e, "inputGrade2")}
+                    onChangeCallback={(e) =>
+                        this.handleChange(e, "inputGrade2")
+                    }
+                    required={true}
+                    description="Enter the higher grade"
+                    validators={[
+                        {
+                            validate: (grade2) => grade2 != "",
+                            message: "You must input a grade",
+                        },
+                        {
+                            validate: (grade2) => parseInt(grade2) <= 12,
+                            message: "Grade cannot be greater than 12",
+                        },
```

Comment:
We don't have to get too crazy with these validators since we do a lot of checking in the backend. But, we can add another easy one here.

```
{
         validate: (grade2) => this.state.inputGrade1 < parseInt(grade2),
         message: "Grade2 must be greater than Grade1",
},
```